go/pkg/filemetadata/filemetadata.go

```diff
@@ -30,6 +31,32 @@ type FileError struct {
 	Err        error
 }
 
+// External xattr package can be mocked for testing through this interface
+type xattributeAccessorInterface interface {
+	isSupported() bool
+	getXAttr(path string, name string) ([]byte, error)
+}
+
+type xattributeAccessor struct{}
+
+func (x xattributeAccessor) isSupported() bool {
+	return xattr.XATTR_SUPPORTED
+}
+
+func (x xattributeAccessor) getXAttr(path string, name string) ([]byte, error) {
+	return xattr.Get(path, name)
+}
+
+var (
+	XattrDigestName string
+	XattrAccess     xattributeAccessorInterface
+)
+
+func init() {
+	XattrDigestName = ""
```

Comment:
In my reclient code I've got 
`flag.StringVar(&filemetadata.XattrDigestName, "xattr_hash", "", "Extended file attribute to obtain the hash from, if available")`
Do we want this to be on by default?  Because that's how I've chosen to configure it: if the name is there, attempt to use it, otherwise skip to the full file hashing.  If it gets a default value then it'll need to actively be turned off.

Reply:
> Do we want this to be on by default? Because that's how I've chosen to configure it: if the name is there, attempt to use it, otherwise skip to the full file hashing. If it gets a default value then it'll need to actively be turned off.

Ah right, no we don't want it to be on by default. What you have in re-client sgtm.